server/src/main/java/org/elasticsearch/action/search/SearchResponseMerger.java

```diff
@@ -118,8 +122,8 @@ int numResponses() {
     public SearchResponse getMergedResponse(Clusters clusters) {
         // if the search is only across remote clusters, none of them are available, and all of them have skip_unavailable set to true,
         // we end up calling merge without anything to merge, we just return an empty search response
-        if (searchResponses.size() == 0) {
-            return SearchResponse.empty(searchTimeProvider::buildTookInMillis, clusters);
+        if (searchResponses.isEmpty()) {
+            return SearchResponse.empty(() -> searchResponseMetrics.recordTookTime(searchTimeProvider.buildTookInMillis()), clusters);
```

Comment:
I find it weird that SearchResponse.empty calls withoutMetrics, yet in this case the took time is recorded, but externally. Also, don't you build took in millis twice, hence the time you record is different from the one the search response holds, after all?

This is a weird edge-case where we create an artificial search response with no results. I wonder how significant it is to record its took time, because it does nothing. Yet technically we do compute a took time that we expose to users.